### ISSUE TITLE
Unable to get the checkpoint through training

### ISSUE THREAD

**Issue by goirik-chakrabarty** [jsu27/decomp_diffusion#1]
Dear authors,

Thank you for the code. It works seamlessly, and it is easy to understand. However, I cannot run the training script and get a good decomposition. However, the ckpt you have provided is working for me. 

Can you please help me train the model so that I can do good decomposition on the CLEVR dataset? Is there some particular seed value for which it works?

Regards,

**Comment by jsu27** [comment]
Hi Goirik, in our experiments, a run with batch size 32 and 10000 steps could achieve good decomposition quality for CLEVR. How long did you train the model for?

**Comment by Yanyi-Zhang** [comment]
> Hi Goirik, in our experiments, a run with batch size 32 and 10000 steps could achieve good decomposition quality for CLEVR. How long did you train the model for?

Hi Jocelin, 

Thank you for sharing your work and settings. I followed your configuration to train the model but also couldn’t achieve good decomposition results. Especially, when generating images from individual learned representations, the outputs were not satisfactory. I encountered the same issue when using the CelebA-HQ dataset. Could you please share more detailed hyperparameter settings?

Additionally, I couldn’t find the testing dataset or the code for quantitative evaluation. Could you please provide those as well?

Thank you for your help!

**Comment by jsu27** [comment]
Hi Yanyi, certainly for CLEVR, the above settings & default parameters should be sufficient. Could you describe what was unsatisfactory, eg was the individual component empty or blurry? Were you able to get good decompositions for Tetris, a simpler dataset?

In addition, the quantitative evaluation for FID/KID and LPIPS are here: https://github.com/jsu27/decomp_diffusion/issues/3#issuecomment-2560579542. The method is evaluated on reconstruction of learned factors from the training dataset, so the same dataset is used in evaluation.

**Comment by Yanyi-Zhang** [comment]
Hi Jocelin,

Thank you for your reply. I trained the model using the settings and default hyperparameters you provided. I’ve attached a few sample results that I randomly selected from the dataset.

It seems that the reconstructed images do not fully match the input images, and the outputs generated from individual components don’t capture the local or global factors very clearly. In the attached figure, the leftmost column shows the input images, the rightmost column shows the reconstructed results, and the middle four columns correspond to the images generated by individual components.

<img width="1100" height="397" alt="Image" src="https://github.com/user-attachments/assets/d0c83d6d-b783-4b27-ba96-a31d2ca9a03f" />

<img width="795" height="463" alt="Image" src="https://github.com/user-attachments/assets/18cbcbd7-9c16-45e3-9cb6-ae06bb0f084d" />
